streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -812,7 +812,8 @@ private void runLoop() {
                         "This implies that this thread missed a rebalance and dropped out of the consumer group. " +
                         "Will try to rejoin the consumer group. Below is the detailed description of the task:\n{}",
                     ignoreAndRejoinGroup.migratedTask().id(), ignoreAndRejoinGroup.migratedTask().toString(">"));
-
+                log.info("Resetting numIterations from {} to 1", numIterations);
```

Comment:
So we need this log? We already warn a WARN and `numIteration` is an internal detail that I think we should not "leak" into the logs.

Reply:
+1.

Also, could we add a unit test for this case?

Reply:
Also note, that txn timeout is one of the reasons of `TaskMigratedException` only, so the title seems inconsistent with the actual change here. If we indeed intent to reset for any `TaskMigratedException` root causes, then we should change the title of the PR; otherwise, we should get the cause of the exception and check if it is `ProducerIdFenced` exception and only reset then.

Reply:
Ack.  Agreed, I'll do the latter as that seems to make the most sense to me.